core/httpd.c

```diff
@@ -850,7 +850,7 @@ CallbackStatus ICACHE_FLASH_ATTR httpdRecvCb(HttpdInstance *pInstance, HttpdConn
                     httpdProcessRequest(pInstance, conn);
                 }
             }
-        } else if (conn->post.len!=0) {
+        } else if (conn->post.buff && conn->post.len!=0) {
```

Comment:
when does this condition occur?

Reply:
like, is it possible to get a non-zero length without a buffer? I guess when we run out of buffers?

Reply:
On esp8266 the resource is very limited. As a result, the malloc() on post.buff may return null, but nothing e.g. flags will indicate this. Then a violate access happen.